prov/efa/src/efa_hmem.c

```diff
@@ -85,26 +99,40 @@ static int efa_hmem_support_status_update_cuda(struct efa_hmem_support_status *c
 		return ret;
 	}
 
-	cuda_status->p2p_supported = true;
+	cuda_info->p2p_supported = true;
+	/* Eager and runting read protocols */
+	if (-FI_ENODATA != fi_param_get(&rxr_prov, "inter_max_medium_message_size", &tmp_value)) {
+		EFA_WARN(FI_LOG_DOMAIN,
+		         "The environment variable FI_EFA_INTER_MAX_MEDIUM_MESSAGE_SIZE was set, "
+		         "but EFA HMEM via Cuda API only supports eager and runting read protocols. Please "
+		         "unset the variable and run again.\n");
+	}
+
+	if (-FI_ENODATA == fi_param_get(&rxr_prov, "inter_min_read_message_size", &tmp_value)) {
+		cuda_info->min_read_msg_size = efa_max_eager_msg_size_with_largest_header(efa_domain) + 1;
+		cuda_info->min_read_write_size = efa_max_eager_msg_size_with_largest_header(efa_domain) + 1;
+	}
```

Comment:
should not here be an else branch that make `cuda_info->min_read_msg_size` to be same as `tmp_value`?

Reply:
tmp value was already defaulted so code did not have to be copy/pasted

Reply:
I would like the logic of handling the value in one place. Can we put the logic in one helper function and let different hmem call them?

Reply:
done